File: src/main/kotlin/main/structures.kt
package main

import java.time.LocalDate
import java.time.ZonedDateTime

data class UserDay(val date: LocalDate, val answers: List<Answer>)

data class Answer(val subject: String, val rating: Int, val date: ZonedDateTime)
data class UserAnswer(val user: Int, val answer: Answer)


data class UserActivity(val user: Int, val days: List<UserDay>) {
    fun answersCount(): Int {
        return days.map { it.answers.size }.sum()
    }

    fun avgRating(): Double {
        return days.flatMap { it.answers }
            .map { it.rating }
            .average()
    }
}

File: main/users/structures.go
package users

import "time"

type UserAnswer struct {
	userID  int
	subject string
	rating  int
	date    time.Time
}

type UserDay struct {
	date    string
	answers []UserAnswer
}

type UserActivity struct {
	userID int
	days   []UserDay
}

func (activity UserActivity) answersCount() int {
	sum := 0
	for _, x := range activity.days {
		sum += len(x.answers)
	}
	return sum
}

func (activity UserActivity) averageRating() float64 {
	sum := 0.0
	count := 0.0
	for _, day := range activity.days {
		for _, answer := range day.answers {
			sum += float64(answer.rating)
			count += 1
		}
	}
	f := sum / count
	return f
}
